dialog/base.go

```diff
@@ -37,19 +38,38 @@ type dialog struct {
 	dismiss        *widget.Button
 
 	response  chan bool
+	mu        sync.Mutex
 	responded bool
 	parent    fyne.Window
 }
 
+func (d *dialog) setupWait() {
+	d.mu.Lock()
+	if d.response != nil {
+		// Already shown
```

Comment:
A comment in code probably indicates it could be clearer. How about a function `isShown` that checks (with a lock) the value of `d.response`? Calling that would remove the need for this, and the later (inverted) case, comments.